src/apps/activities/crud/activity.py

```diff
@@ -55,9 +58,44 @@ async def get_mobile_with_items_by_applet_id(self, applet_id: uuid.UUID, is_revi
             ActivitySchema.order,
             ActivitySchema.scores_and_reports,
         )
+
+        # Query filters
+        filters = query_params.filters
+
+        if filters.get("has_score") is not None:
+            query = query.join(
+                ActivityItemSchema,
+                ActivityItemSchema.activity_id == ActivitySchema.id,
+            )
+
         query = query.where(ActivitySchema.applet_id == applet_id)
         if isinstance(is_reviewable, bool):
             query = query.where(ActivitySchema.is_reviewable == is_reviewable)
+
+        if filters.get("has_submitted") is not None:
+            has_submitted = filters.get("has_submitted")
+            if isinstance(has_submitted, bool):
+                query = query.where(
+                    text(
+                        """%s (SELECT 1 FROM answers WHERE 
```

Comment:
You cannot use answers and answer_items tables with other ones in one query because of arbitrary server. Please use `answer_session` in AnswerService

Reply:
@vshvechko Not sure I understood here. `answer_session` is not a table right? (Or maybe my DB is outdated) How can I use this to filter activities that have answers? Can you provide an example?

Reply:
`answer_session` is separate sqlalchemy session for querying `answers` and `answer_items` tables. Just search `get_asnwer_session` dependency in the app. E.g. src.apps.answers.api.create_answer()